vcf.c

```diff
@@ -483,7 +575,16 @@ bcf_hrec_t *bcf_hdr_parse_line(const bcf_hdr_t *h, const char *line, int *len)
         if (bcf_hrec_set_val(hrec, hrec->nkeys-1, p, r-p, quoted) < 0)
             goto fail;
         if ( quoted && *q==ending ) q++;
-        if ( *q=='>' ) { nopen--; q++; }
+        if ( *q=='>' )
+        {
+            if (nopen) nopen--;     // this can happen with nested angle brackets <>
+            q++;
+        }
+    }
+    if ( nopen && warn_unfinished_line )
+    {
+        warn_unfinished_line = 0;
```

Comment:
Part of me feels the guard for whether we've warned once already isn't necessary.

If I have dozens of invalid lines in my file, it's not really very helpful to get a fault reported, fix it, run the software and get the next line reported, fix, repeat...

Besides, frankly if people have hundreds and hundreds of badly formatted headers then beating them around the head repeatedly for each and every one may be considered to be of benefit to the community. :-)